File: 0xxx/02xx/021x/0219.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
const containsNearbyDuplicate = function (nums, k) {
  const hash = new Map()

  for (let i = 0; i < nums.length; i++) {
    if (hash.has(nums[i]) && i - hash.get(nums[i]) <= k) {
      return true
    } else {
      hash.set(nums[i], i)
    }
  }
  return false
}

File: 0xxx/02xx/021x/0219.py
class Solution:
  def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
    recorder = {}
    for i, num in enumerate(nums):
      if num in recorder and (i - recorder[num] <= k):
        return True
      else:
        recorder[num] = i
    return False